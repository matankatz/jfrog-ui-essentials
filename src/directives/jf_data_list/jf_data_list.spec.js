import {items} from './jf_data_list.mock';

describe('unit test: jf_table_view directive & JFTableViewOptions service', () => {

    var $scope;
    var $rootScope;
    var testAppScope;
    var $q;
    var $timeout;
    var JFrogTableViewOptions;
    let $elem;

    const setup = (_$timeout_, _$q_, _$rootScope_) => {
        $timeout = _$timeout_;
        $rootScope = _$rootScope_;
        $q = _$q_;
    }

    const compileDirective = (attr) => {
        ({$scope, $elem} = window.compileDirectiveAndGetElement('jf-data-list', attr));
        // = compileResult.$scope;
        // $elem = $(compileResult.elem);
        $scope.$apply();
    }

    beforeEach(angular.mock.module('jfrog.ui.essentials'));
    beforeEach(inject(setup));

    beforeEach(() => {
        compileDirective({
            items,
        });
    });

    it('should compile', function (done) {
        expect($elem[0].outerHTML).toMatchSnapshot();
        done();
    });


});