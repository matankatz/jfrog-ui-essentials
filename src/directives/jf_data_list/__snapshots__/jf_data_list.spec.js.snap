// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test: jf_table_view directive & JFTableViewOptions service should compile 1`] = `
"<jf-data-list items=\\"data.items\\" class=\\"ng-scope ng-isolate-scope\\"><table class=\\"data-list\\">
	<tbody><!-- ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Plain text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">nam nemo nesciunt omnis perferendis possimus</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">More plain text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">Asperiores at consectetur consequatur consequuntur cumque eos eum</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Text with copy to clipboard:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">optio possimus provident quas quos</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) --><div class=\\"copy ng-scope\\" ng-if=\\"item.copy &amp;&amp; !jfDataList.isArray(item.value)\\">
					<jf-clip-copy text-to-copy=\\"item.value\\" class=\\"ng-isolate-scope\\"><span class=\\"clip-copy-button\\">
    <a ng-show=\\"!jfClipCopy.isSafari &amp;&amp; !!jfClipCopy.textToCopy\\" class=\\"copy-to-clip icon icon-copy-to-clipboard-2\\" data-clipboard-text=\\"{{jfClipCopy.textToCopy}}\\" jf-tooltip=\\"{{jfClipCopy.tooltipText}}\\"></a>
    <a ng-show=\\"jfClipCopy.isSafari &amp;&amp; !!jfClipCopy.textToCopy\\" class=\\"copy-to-clip icon icon-copy-to-clipboard\\" clip-copy=\\"jfClipCopy.textToCopy\\" clip-click=\\"jfClipCopy.onSuccessfulCopy()\\" jf-tooltip=\\"{{jfClipCopy.tooltipText}}\\"></a>
</span>

</jf-clip-copy>
				</div><!-- end ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Overflowing text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Accusantium, adipisci architecto atque delectus doloremque eos eum expedita explicabo illo ipsa itaque iusto nam neque perferendis, repellendus soluta vitae voluptatibus. Eos est hic id perspiciatis quasi? Ipsa magni optio quod?</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Overflowing text with copy:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">dolore eius error periores at consectetur consequuntur cumque eos eum nam nemo nesciunt omnis perferendis fugiat harum availablePackageTypesnesciunt perferendis fugiat harum, impedit ipsum iste</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) --><div class=\\"copy ng-scope\\" ng-if=\\"item.copy &amp;&amp; !jfDataList.isArray(item.value)\\">
					<jf-clip-copy text-to-copy=\\"item.value\\"></jf-clip-copy>
				</div><!-- end ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Empty text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">&nbsp;</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">List with some clickable items:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) --><div ng-if=\\"jfDataList.isArray(item.value)\\" id=\\"data-list-row-6\\" class=\\"ng-scope\\">
					<!-- ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com\\">Lorem</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url -->
						<!-- ngIf: !tag.url --><span class=\\"gridcell-content-text ng-binding ng-scope\\" ng-if=\\"!tag.url\\" ng-bind-html=\\"tag.label\\">Accusantium</span><!-- end ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url -->
						<!-- ngIf: !tag.url --><span class=\\"gridcell-content-text ng-binding ng-scope\\" ng-if=\\"!tag.url\\" ng-bind-html=\\"tag.label\\">adipisci</span><!-- end ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">Ipsum</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">Dolor</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url -->
						<!-- ngIf: !tag.url --><span class=\\"gridcell-content-text ng-binding ng-scope\\" ng-if=\\"!tag.url\\" ng-bind-html=\\"tag.label\\">architecto</span><!-- end ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value -->
					<!-- ngIf: item.value.length >= 1 && jfDataList.htmlIsOverflowing('#data-list-row-' + $parent.$index) -->
				</div><!-- end ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">List with overflowing items:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) --><div ng-if=\\"jfDataList.isArray(item.value)\\" id=\\"data-list-row-7\\" class=\\"ng-scope\\">
					<!-- ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com\\">Lorem</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">Ipsum</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">Dolor</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url -->
						<!-- ngIf: !tag.url --><span class=\\"gridcell-content-text ng-binding ng-scope\\" ng-if=\\"!tag.url\\" ng-bind-html=\\"tag.label\\">Accusantium</span><!-- end ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url -->
						<!-- ngIf: !tag.url --><span class=\\"gridcell-content-text ng-binding ng-scope\\" ng-if=\\"!tag.url\\" ng-bind-html=\\"tag.label\\">adipisci</span><!-- end ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url -->
						<!-- ngIf: !tag.url --><span class=\\"gridcell-content-text ng-binding ng-scope\\" ng-if=\\"!tag.url\\" ng-bind-html=\\"tag.label\\">architecto</span><!-- end ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">Sit</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">Amet</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">Consectetur</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"https://www.lipsum.com/\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"https://www.lipsum.com/\\">elit</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"http://www.google.com\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"http://www.google.com\\">atque</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"http://www.google.com\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"http://www.google.com\\">delectus</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value --><div class=\\"tag ng-scope\\" ng-repeat=\\"tag in item.value\\">
						<!-- ngIf: tag.url --><a class=\\"gridcell-content-text jf-link ng-binding ng-scope\\" ng-if=\\"tag.url\\" ng-href=\\"http://www.google.com\\" target=\\"_blank\\" ng-bind-html=\\"tag.label\\" href=\\"http://www.google.com\\">doloremque</a><!-- end ngIf: tag.url -->
						<!-- ngIf: !tag.url -->
					</div><!-- end ngRepeat: tag in item.value -->
					<!-- ngIf: item.value.length >= 1 && jfDataList.htmlIsOverflowing('#data-list-row-' + $parent.$index) -->
				</div><!-- end ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Value is url:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) --><a ng-if=\\"item.isUrl &amp;&amp; item.value &amp;&amp; item.value.length &amp;&amp; ((!item.hasOwnProperty('url')) || (item.url &amp;&amp; item.url.length))\\" ng-href=\\"http://www.lorem_ipsum_dolor_sit_amet_consectetur_adipisicing_elit.com\\" ng-bind-html=\\"item.value\\" target=\\"_blank\\" class=\\"jf-link ng-binding ng-scope ng-isolate-scope\\" jf-tooltip-on-overflow=\\"\\" href=\\"http://www.lorem_ipsum_dolor_sit_amet_consectetur_adipisicing_elit.com\\">http://www.lorem_ipsum_dolor_sit_amet_consectetur_adipisicing_elit.com</a><!-- end ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Value with separate url:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) --><a ng-if=\\"item.isUrl &amp;&amp; item.value &amp;&amp; item.value.length &amp;&amp; ((!item.hasOwnProperty('url')) || (item.url &amp;&amp; item.url.length))\\" ng-href=\\"https://www.lipsum.com/\\" ng-bind-html=\\"item.value\\" target=\\"_blank\\" class=\\"jf-link ng-binding ng-scope ng-isolate-scope\\" jf-tooltip-on-overflow=\\"\\" href=\\"https://www.lipsum.com/\\">Link to lorem ipsum website</a><!-- end ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Empty url:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">Link to lorem ipsum website</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Empty url &amp; text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\">&nbsp;</div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems -->
</tbody></table></jf-data-list>"
`;
